File: output/ K 个一组翻转链表/ K 个一组翻转链表.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */

struct ListNode* reverseKGroup(struct ListNode* head, int k){
    struct ListNode* newChain,*start,*next_start=NULL,*next,*p,*target;
    int len = 1;//链表长度
    int loop;//需要反转的片段数量
    //当k为1时直接返回原链表
    if(k==1){
        return head;
    }
    //获取新的表头
    p = head;
    for(int i=0;i<k-1;i++){
        p = p->next;
    };
    newChain = p;
    
    //获取链表长度
    next = head->next;
    while(next!=NULL){
        len ++;
        next = next->next;
    };
    //获取要循环的次数
    loop = len/k;
    start = head;
    next = newChain;
    while(loop+1>0){
        if(loop>0){
            p = start;
            for(int i=0;i<k;i++){
                p = p ->next;
            };
            next_start = p;   
        }
        int n2 = k-1;
        if(loop>0){
            for(int n=k;n>0;n--){
                target = start;
                int index = n2;
                while(index>0){
                    target = target->next;//target
                    index--;
                };
                n2 --;
                if(next!=target){
                    next->next = target;
                }
                next = target;
            };
        }else{
            next->next = start;
        }
        start = next_start;
         loop--;
    };
    return newChain;
}

File: output/ K 个一组翻转链表/ K 个一组翻转链表.rs
#[derive(PartialEq, Eq, Clone, Debug)]
pub struct ListNode {
    pub val: i32,
    pub next: Option<Box<ListNode>>,
}

impl ListNode {
    #[inline]
    fn new(val: i32) -> Self {
        ListNode {
            next: None,
            val,
        }
    }
}

pub fn reverse_k_group(head: Option<Box<ListNode>>, k: i32) -> Option<Box<ListNode>> {
    if k == 1 {
        return head;
    }
    
    let head = head?;
    
    // Get length of the list
    let mut len = 1;
    let mut current = &head;
    while let Some(ref next_node) = current.next {
        len += 1;
        current = next_node;
    }
    
    // Number of groups to reverse
    let loop_count = len / k;
    if loop_count == 0 {
        return Some(head);
    }
    
    // Convert to vector for easier manipulation
    let mut nodes = Vec::new();
    let mut current = Some(head);
    while let Some(mut node) = current {
        current = node.next.take();
        nodes.push(node);
    }
    
    // Reverse groups
    for group in 0..loop_count {
        let start_idx = (group * k) as usize;
        let end_idx = start_idx + k as usize;
        nodes[start_idx..end_idx].reverse();
    }
    
    // Rebuild linked list
    for i in (0..nodes.len() - 1).rev() {
        nodes[i].next = Some(nodes.remove(i + 1));
    }
    
    Some(nodes.into_iter().next().unwrap())
}